### PR TITLE
Studio sample （update）

### PR DESCRIPTION
**Pull request by 851996006**

### CODE REVIEW

File: sample_analyze_layout.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+FILE: sample_analyze_layout.py
+
+DESCRIPTION:
+    This sample demonstrates how to extract text, tables, selection marks, and layout information from a document
+    given through a file.
+
```
Comment (luzhang06) [inline sample_analyze_layout.py]: text, tables, figures and selection marks and document structure (e.g., sections) information from a document.

File: sample_analyze_layout.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+"""
+FILE: sample_analyze_layout.py
+
+DESCRIPTION:
+    This sample demonstrates how to extract text, tables, selection marks, and layout information from a document
+    given through a file.
+    
+    Note that selection marks returned from begin_analyze_document(model_id="prebuilt-layout") do not return the text
+    associated with the checkbox. For the API to return this information, build a custom model to analyze the
+    checkbox and its text. See sample_build_model.py for more information.
```
Comment (luzhang06) [inline sample_analyze_layout.py]: For this part, recommend moving to selection mark extraction comments part. Not necessary to show up at the beginning of the sample.

File: sample_analyze_layout.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+PREREQUISITES:
+    Necessary prerequisites are listed below.
+    To find more details, please click the "How-to guides" link (https://aka.ms/AApjann) to visit the Documentation.
```
Comment (luzhang06) [inline sample_analyze_layout.py]: the short link seems not pointing to the section, can you please double check?

File: sample_analyze_layout.py [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+    -------Python and IDE------
+    1) Python3.7 or latter (https://www.python.org/) . Your Python installation should include pip (https://pip.pypa.io/en/stable/).
```
Comment (luzhang06) [inline sample_analyze_layout.py]: remove additional space

File: sample_analyze_layout.py [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+    formUrl = "https://raw.githubusercontent.com/Azure-Samples/cognitive-services-REST-api-samples/master/curl/form-recognizer/sample-layout.pdf"
+    # Replace with the your actual formUrl. To find more URLs of sample documents，visit: https://aka.ms/AApj4dy 
+    poller = document_intelligence_client.begin_analyze_document_from_url(
+        "prebuilt-layout",formUrl
+    )        
+    
+    # Analyze a local document：
+    # Delete or comment out the part of "Analyze a document at a URL" above.
+    # Uncomment the following codes. Enter the actual file path in "path_to_sample_documents" variable.
+    # path_to_sample_documents = os.path.abspath(
```
Comment (luzhang06) [inline sample_analyze_layout.py]: path_to_sample_document - this should be only one document, we don't support multiple documents as input in an API call

File: sample_analyze_layout.py [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+            if table.bounding_regions:
+                for region in table.bounding_regions:
+                    print(f"Table # {table_idx} location on page: {region.page_number} is {region.polygon}")
+            # Analyze cells.
+            for cell in table.cells:
+                print(f"...Cell[{cell.row_index}][{cell.column_index}] has text '{cell.content}'")
+                if cell.bounding_regions:
+                    for region in cell.bounding_regions:
+                        print(f"...content on page {region.page_number} is within bounding polygon '{region.polygon}'")
+
```
Comment (luzhang06) [inline sample_analyze_layout.py:159]: per our offline discussion, please add "extract detected figures" sample code here as well.

File: sample_analyze_layout.py [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+                print(f"Received an invalid image error: {error.error}")
+            if error.error.code == "InvalidRequest":
+                print(f"Received an invalid request error: {error.error}")
+            # Raise the error again after printing it
+            raise
+        # If the inner error is None and then it is possible to check the message to get more information:
+        if "Invalid request".casefold() in error.message.casefold():
+            print(f"Uh-oh! Seems there was an invalid request: {error}")
+        # Raise the error again
+        raise
```
Comment (luzhang06) [inline sample_analyze_layout.py:193]: Put some comments for next step, we can provide 2 links:
- Learn more about Layout model: https://aka.ms/di-layout
- Find more sample code: https://aka.ms/doc-intelligence-samples

File: sample_analyze_read.py [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+    )       
+    # # If analyzing a local document, remove the comment markers (#) at the beginning of these 11 lines.
+    # # Delete or comment out the part of "Analyze a document at a URL" above.
+    # # Replace <path to your sample file>  with your actual file path.
+    # path_to_sample_document = "<path to your sample file>"
+    # with open(path_to_sample_document, "rb") as f:
+    #     poller = document_intelligence_client.begin_analyze_document(
+    #         "prebuilt-read",
+    #         analyze_request=f,
+    #         features=[DocumentAnalysisFeature.STYLE_FONT],
```
Comment (luzhang06) [inline sample_analyze_read.py]: There's addon feature turned on here, optional, you need to either delete it or make some comments on it for addon capability；）

File: sample_analyze_read.py [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+    #         content_type="application/octet-stream",
+    #     )
+    result: AnalyzeResult = poller.result()
+    
+    # [START analyze_read]
+    # Detect languages.
+    print("----Languages detected in the document----")
+    if result.languages is not None:
+        for language in result.languages:
+            print(f"Language code: '{language.locale}' with confidence {language.confidence}")
```
Comment (luzhang06) [inline sample_analyze_read.py:118]: Based on these lines, recommend turn on language addon in the previous comment.

File: sample_analyze_read.py [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+
+    # Detect styles.
+    print("----Styles detected in the document----")
+    if result.styles:
+        for style in result.styles:
+            if style.is_handwritten:
+                print("Found the following handwritten content: ")
+                print(",".join([result.content[span.offset : span.offset + span.length] for span in style.spans]))
+            if style.font_style:
```
Comment (luzhang06) [inline sample_analyze_read.py]: Can you please help verify if free tier resource can use paid addon?

File: sample_analyze_read.py [hunk 11]
```diff
@@ -0,0 +1,192 @@
...
+                words = get_words(page, line)
+                print(
+                    f"...Line # {line_idx} has {len(words)} words and text '{line.content}' within bounding polygon '{line.polygon}'"
+                )
+
+                # Analyze words.
+                for word in words:
+                    print(f"......Word '{word.content}' has a confidence of {word.confidence}")
+
+        # Analyze selection marks.
```
Comment (luzhang06) [inline sample_analyze_read.py]: Read model should not have selection marks. It's only available in Layout and some prebuilt model.

File: sample_analyze_read.py [hunk 12]
```diff
@@ -0,0 +1,192 @@
...
+                    f"...Selection mark is '{selection_mark.state}' within bounding polygon "
+                    f"'{selection_mark.polygon}' and has a confidence of {selection_mark.confidence}"
+                )
+        # Note that selection marks returned from begin_analyze_document(model_id="prebuilt-read") do not return the text associated with the checkbox. 
+        # For the API to return this information, build a custom model to analyze the checkbox and its text. For detailed steps, visit: https://aka.ms/train-your-custom-model
+        
+    # Analyze paragraphs.
+    if result.paragraphs:
+        print(f"----Detected #{len(result.paragraphs)} paragraphs in the document----")
+        for paragraph in result.paragraphs:
```
Comment (luzhang06) [inline sample_analyze_read.py:141]: can you please double check if paragraph.role applies to read model output? My understanding is this only applies to layout model.

File: sample_analyze_receipts.py [hunk 13]
```diff
@@ -0,0 +1,184 @@
...
+    # # Replace <path to your sample file>  with your actual file path.
+    # path_to_sample_document = "<path to your sample file>"
+    # with open(path_to_sample_document, "rb") as f:
+    #     poller = document_intelligence_client.begin_analyze_document(
+    #         "prebuilt-receipt", analyze_request=f, locale="en-US", content_type="application/octet-stream"
+    #     )
+    receipts: AnalyzeResult = poller.result()
+
+    # [START analyze_receipts]
+    # To find more schemas, please refer to: https://aka.ms/azsdk/documentintelligence/receiptfieldschema
```
Comment (luzhang06) [inline sample_analyze_receipts.py]: this link seems not work any more, should be something like this: https://learn.microsoft.com/en-us/azure/ai-services/document-intelligence/concept-receipt?view=doc-intel-4.0.0#field-extraction

File: sample_analyze_tax_us_w2.py [hunk 14]
```diff
@@ -0,0 +1,324 @@
...
+
+"""
+FILE: sample_analyze_tax_us_w2.py
+
+DESCRIPTION:
+    This sample demonstrates how to analyze US W-2 tax forms.
+
+    See fields found on a US W-2 tax form here:
+    https://aka.ms/azsdk/documentintelligence/taxusw2fieldschema
+
```
Comment (luzhang06) [inline sample_analyze_tax_us_w2.py:17]: Same, the short links seem not work. Please double check. Also it's a good idea to put the field schema link in the description section, please apply it to other prebuilt models as well.

File: sample_analyze_read.py (hunk 12 above)
Comment (851996006) [reply sample_analyze_read.py:141]: https://learn.microsoft.com/en-us/azure/ai-services/document-intelligence/concept-read?view=doc-intel-4.0.0#paragraphs
The Read concept page shows the extraction of the paragraph, and the output of the code is the same as the "output" display in the document.

File: sample_analyze_read.py (hunk 10 above)
Comment (851996006) [reply sample_analyze_read.py]: I asked customer service, and the feedback was: some require upgrading and paying to use it, and the free ones also have limits.
image
image

File: sample_analyze_read.py (hunk 8 above)
Comment (851996006) [reply sample_analyze_read.py]: Is my understanding correct?
image

File: sample_analyze_read.py (hunk 11 above)
Comment (851996006) [reply sample_analyze_read.py]: Already deleted

File: sample_analyze_receipts.py (hunk 13 above)
Comment (851996006) [reply sample_analyze_receipts.py]: Replaced with field-related links and placed field schema links in the description section.

File: sample_analyze_tax_us_w2.py (hunk 14 above)
Comment (851996006) [reply sample_analyze_tax_us_w2.py:17]: Already applied it to other prebuilt models

**Pull request merged**